packages/core/src/validate/overrides.ts

```diff
@@ -5,67 +5,68 @@ import { ValidationError } from './run';
+  'inline-assembly': `    You are using the \`unsafeAllow.inline-assembly\` flag.\n`,
+};
 
-export function processExceptions(
-  contractName: string,
-  errorsToProcess: ValidationError[],
-  opts: ValidationOptions,
-): ValidationError[] {
-  const { unsafeAllowCustomTypes, unsafeAllowLinkedLibraries } = withValidationDefaults(opts);
-  let errors: ValidationError[] = errorsToProcess;
+export function withValidationDefaults(opts: ValidationOptions): Required<ValidationOptions> {
+  const unsafeAllowCustomTypes = opts.unsafeAllowCustomTypes ?? false;
+  const unsafeAllowLinkedLibraries = opts.unsafeAllowLinkedLibraries ?? false;
+  const unsafeAllow = opts.unsafeAllow ?? [];
 
-  // Process `unsafeAllowCustomTypes` flag
   if (unsafeAllowCustomTypes) {
-    errors = processOverride(
-      contractName,
-      errors,
-      ['enum-definition', 'struct-definition'],
-      `    \`unsafeAllowCustomTypes\` may not be necessary.\n` +
-        `    Update your plugin for automated struct and enum checks.\n`,
-    );
+    unsafeAllow.push('enum-definition', 'struct-definition');
   }
-
-  // Process `unsafeAllowLinkedLibraries` flag
   if (unsafeAllowLinkedLibraries) {
-    errors = processOverride(
-      contractName,
-      errors,
-      ['external-library-linking'],
-      `    You are using the \`unsafeAllowLinkedLibraries\` flag to include external libraries.\n` +
-        `    Make sure you have manually checked that the linked libraries are upgrade safe.\n`,
-    );
+    unsafeAllow.push('external-library-linking');
   }
 
-  return errors;
+  return { unsafeAllowCustomTypes, unsafeAllowLinkedLibraries, unsafeAllow };
 }
 
-function processOverride(
+export function processExceptions(
   contractName: string,
-  errorsToProcess: ValidationError[],
-  overrides: string[],
-  message: string,
+  errors: ValidationError[],
+  opts: ValidationOptions,
 ): ValidationError[] {
-  let errors: ValidationError[] = errorsToProcess;
-  let exceptionsFound = false;
+  const { unsafeAllow } = withValidationDefaults(opts);
 
-  errors = errors.filter(error => {
-    const isException = overrides.includes(error.kind);
-    exceptionsFound = exceptionsFound || isException;
-    return !isException;
-  });
+  for (const [errorType, errorDescription] of Object.entries(ValidationErrorUnsafeMessages)) {
+    if (unsafeAllow.includes(errorType as ValidationError['kind'])) {
+      let exceptionsFound = false;
 
-  if (exceptionsFound && !silenced) {
-    console.error(
-      chalk.keyword('orange').bold('Warning:') + ` Potentially unsafe deployment of ${contractName}\n\n` + message,
-    );
+      errors = errors.filter(error => {
+        const isException = errorType === error.kind;
+        exceptionsFound = exceptionsFound || isException;
+        return !isException;
+      });
+
+      if (exceptionsFound && !silenced && errorDescription) {
+        console.error(
+          '\n' +
```

Comment:
I think I've removed these initial newlines before, I think it looks better without them.